src/main/webapp/app/localci/build-queue/build-queue.component.html

```diff
@@ -412,12 +412,34 @@ <h3 id="build-queue-queued-heading" jhiTranslate="artemisApp.buildQueue.queuedBu
     }
 </div>
 <div>
-    <div style="display: flex; align-items: center">
-        <h3 id="build-queue-finished-heading" jhiTranslate="artemisApp.buildQueue.finishedBuildJobs" style="margin-right: 20px"></h3>
-        <button class="btn btn-primary" (click)="refresh()">
-            <fa-icon [icon]="faSync" />
-            <span jhiTranslate="metrics.refresh.button"></span>
-        </button>
+    <div class="d-flex justify-content-between">
+        <div style="display: flex; align-items: center">
+            <h3 id="build-queue-finished-heading" jhiTranslate="artemisApp.buildQueue.finishedBuildJobs" style="margin-right: 20px"></h3>
+            <div class="form-group form-inline my-0 mx-1">
+                <input class="form-control ms-3" type="text" name="searchTerm" id="field_searchTerm" [(ngModel)]="searchTerm" (focusout)="triggerLoadFinishedJobs()" />
+                <jhi-help-icon class="ps-1" text="artemisApp.buildQueue.filter.search.tooltip" />
+                <button class="btn btn-primary ms-3" (click)="triggerLoadFinishedJobs()">
+                    <span jhiTranslate="artemisApp.buildQueue.filter.search.title"></span>
+                </button>
+            </div>
+            <button class="btn btn-primary" (click)="loadFinishedBuildJobs()">
+                <fa-icon [icon]="faSync" />
+                <span jhiTranslate="metrics.refresh.button"></span>
+            </button>
+            @if (this.isLoading) {
+                <span class="ms-3" jhiTranslate="artemisApp.buildQueue.filter.search.loading"></span>
+            }
+        </div>
+        <div>
+            <button
+                class="btn"
+                (click)="this.open(content)"
+                [ngClass]="{ 'btn-secondary': !finishedBuildJobFilter.numberOfAppliedFilters, 'btn-success': !!finishedBuildJobFilter.numberOfAppliedFilters }"
```

Comment:
See my other comments about function calls in the template (numberOfAppliedFilters is declared using the `get ...()` syntax)